website/content/en/docs/building-operators/golang/project_migration_guide.md

```diff
@@ -6,177 +6,304 @@ weight: 200
+		IsController: true,
+		OwnerType:    &cachev1alpha1.Memcached{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&cachev1alpha1.Memcached{}).
+		Complete(r)
 }
 ```
 
-2. Update the import aliases in the `Reconcile()` function and copy over any remaining helper functions from `pkg/controller/<kind>/<kind>_controller.go` to `api/<version>/<kind>_controller.go`.
+### Setting the RBAC permissions
+
+The RBAC permissions now are configured via [RBAC markers][rbac_markers] whcih will generate and update the manifest files present in `config/rbac/`. See that on top of the `Reconcile()` method we will find them, in our Memcache example it will be: 
+
+```go
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
+```
+
+However, you might want to configure additional permissions for your project, see [RBAC markers][rbac_markers] doc for more information. 
 
-Refer to [`memcached_controller.go`][kb_memcached_controller] for the example implementation of `Reconcile()` logic which would be present in `api/v1alpha1/memcached_controller.go` of Memcached operator.
+**Note** You  would like to check the final result of our [memcached_controller.go][memcached_controller] migrated to the new layout. 
 
-## Generate CRDs
+## Migrating the Main.go
 
-Run [`make manifests`][generate_crd] to generate CRD manifests. They would be generated inside the `config/crd/bases` folder.
+If you are using third-party API's in your project for example then, it means that you need to copy your customizations from `cmd/manager/main.go` to `main.go`. You’ll also need to ensure all of the needed schemes have been registered.
 
-## Configuring your test environment
+By checking our new `main.go` we will find that:
 
-Projects are scaffolded with unit tests that utilize the [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest)
-library, which requires certain Kubernetes server binaries be present locally.
-Installation instructions can be found [here][env-test-setup].
+- The SDK `leader.Become` was replaced by the [leaderElection][leaderElection]. However, we still able to use the previous one as well by adding the [operator-lib][operator-lib] as dependency fo the project. 
+- The default value for `MetricsBindAddress` is now `8080` instead of `8383`. However, note that it is now configurable via the flag `--metrics-addr`. 
```

Comment:
```suggestion
- The default value for `MetricsBindAddress` is now `8080` instead of `8383`. However, note that it is now configurable via the flag `--metrics-addr`. To continue using port 8383, specify `--metrics-addr=:8383` when you start the operator.
```